### PR TITLE
Upon travelling through a window, a lasers damage is affected by the window itself

### PR DESCRIPTION
**Pull request by MadmanMartian**
:cl:
 * tweak: Lasers going through windows have their damage affected by passing through the window. Should the laser damage reach 1 or below, the laser disperses entirely.

### CODE REVIEW

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
For those gay windows that are actually side window plus fullwindow plus side window does that mean lasers end up at 1/8 strength after going through them

**Comment by Goldsteel** [comment]
I don't think this should apply to regular glass. The linear attenuation coefficient and dispersion for 10cm (generous really) of normal silica glass in the 600nm range gives you a dropoff on the order of km, not cm. Plasma glass, though, sure. Space magic.

**Comment by DrSnips** [comment]
The windows thing is a big concern since a lot of them have the double sided panes which means they lose at least 3/4 of their damage when going through your normal station setup of windows.

**Comment by ghost** [comment]
Correct me if I'm wrong, but this will enable creation of strategically stacked window panes that enable a player to stand inside and be nearly immune to all laser fire, while shooting outside without any penalties.

If you stand inside a turf with a north facing glass pane for example, I do not believe the cross check would be called if you fired north.

**Comment by Zth--** [comment]
+1 this but for plasma windows

**Comment by SonixApache** [comment]
Maybe subtract a quarter or less instead of going straight for the halving.
Also maybe have it per turf rather than per window, considering the above.

**Comment by DeityLink** [comment]
>halving again

jesus christ

**Comment by iFlashYou** [comment]
Better idea would be,
15% reduction for normal glass
30% reduction for reinforced glass
45% reduction for plasma glass
60% reduction for reinforced plasma glass

**Comment by bathosbathtime** [comment]
while /2 seems strong, I still like the general idea of this PR.  I personally wouldn't be against the current state, but a smaller reduction might make it more acceptable to the rest of the server.
More environmental interactions with combat gives room for more tactical play (and abuse but we will burn that bridge when we cross it).

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
What if the damage reduction was capped at 50% i.e. penalty is only applied once to prevent the "whoopsie a window is actually 3 windows" problem
How often do people shoot through 2 windows anyways

**Comment by DrSnips** [comment]
Considering most windows are like this |G| with G being a grille and | being glass panes I'd say pretty often. Then you have triple windows |[G]| where [] is a full tile window this isn't as common as the other but it's still a thing.

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
That's what I meant as the original problem
Let me rephrase as "how often do people shoot through two double/triple windows anyways"

**Comment by ShiftyRail** [comment]
Combine both IFlashYou and Voxperson solutions ; lower reduction cap and cap depending on windows hit
Lasers should also still be useful to destroy windows.

I'm against the "lasers as projectiles" idea for three reasons : 
1/ Immulisons
2/ Lasers make a combat situation slightly more readable
3/ Travel time lasers would just be reskinned ballistics

This solution is better, it forces you to consider your combat environment, allows for tactical cover behind glasses and shit, as well as keeping the 3 points I mentionned.

By the way, lasers can for some reason go through closes firelocks. I think this is part of the laser issue.

**Comment by d3athrow** [comment]
just an idea and opinion
Give various window types a modest(ie not halving damage per window) dispersion coefficient and make laser projectile size determined by its damage and TTL

bonus for silly fun: add ability to make convex/concave windows using welder or something so some madman can make a world conquerin' laser beam with a hallway of 'lenses'

**Comment by MadmanMartian** [comment]
that does sound fun.

I'll add that.

**Comment by MadmanMartian** [comment]
Changes:
 Implemented pomfs suggestion

**Comment by MadmanMartian** [comment]
The size changes is a bit of a bitch due to the way it draws the lines and then deletes them.

Tried it so that the less damage a laser had, the lower its alpha, but that led to lasers not being deleted properly.

**Comment by d3athrow** [comment]
to keep from having to rewrite the beam overlay system could standardize beam widths and include them in the beam master list. ie egun or laser gun damage is the 100% scale standard then it goes up and down from there (cap laser is a bit thicker, pulse rifle or the laser cannon is thicc as heck, etc

probably best for a separate pr though

**Comment by SirSkeleto** [comment]
Does this also apply to glass doors?
